andela/hc-wits-kla
# #158175326 Create standard scheduled tasks

#### What does this PR do?
Allows a user  to create a few standard scheduled tasks that work on any Operating System from a Graphical User Interfaces.
#### Description of Task to be completed?
Users can create scheduled tasks for 
- Backing up a database 
- Send emails to users
- Posting a tweet on twitter
#### How should this be manually tested?
- Clone the repository
- `pip install -r requirements.txt`
- `python manage.py migrate`
-  Install [rabbitmql](https://www.rabbitmq.com/#getstarted)
- Ensure you have postgres client and mysql client because this PR uses `pg_dump` and `mysqldump` to backup databases
- Set `BROKER_URL = 'amqp://localhost'` and `CELERY_RESULT_BACKEND = 'amqp://localhost'` in local_settings.py
- `python manage.py runserver`
-  Navigate to the application url and click the `TASKS` tab 
-  Add a database backup task
- Using a superuser account log into admin center and under Api section add `social media task` and `email task` tasks
- Run the background tasks using the command `celery -A hc worker -l info --beat`
- Note: I will share `TWITTER` and `DROPBOX` keys for development.
-
#### What are the relevant pivotal tracker stories?
[#158175326](https://www.pivotaltracker.com/story/show/158175326)
#### Screenshots (if appropriate)
<img width="1439" alt="screen shot 2018-07-19 at 09 48 50" src="https://user-images.githubusercontent.com/23418080/42943680-1c877190-8b6c-11e8-9762-ae6df0331343.png">
